dask_expr/expr.py

```diff
@@ -1031,6 +1049,32 @@ def _task(self, index: int):
         )
 
 
+class Lengths(Expr):
+    """Returns a tuple of partition lengths"""
+
+    _parameters = ["frame"]
+
+    @property
+    def _meta(self):
+        return tuple()
+
+    def _divisions(self):
+        return (None, None)
+
+    def _simplify_down(self):
+        if isinstance(self.frame, Elemwise):
+            return Lengths(self.frame.operands[0])
```

Comment:
I was simplifying things before.  You'll have to watch out for situations like `x.sum() + x`.  I think that if you look at the implementation for `Len` we handle this well.